src/image/detail/MaskDict.cc

```diff
@@ -53,35 +84,49 @@ class MaskDict::GlobalState {
 
     ~GlobalState() = default;
 
+    std::mutex _mutex;
```

Comment:
I went back and forth on this, not having a ton of experience with either kind of mutex. I went with the non-reentrant mutex because I figured the code that needed to be careful around it was well-bounded (to just this source file).  But I'd be happy to follow a recommendation to go the other way if you think it's better practice in general, and I'll certainly add that warning if I stick with this version.

Reply:
I can't comment on "better practice", it's just what I'm more familiar with. But I do like that it's a) less prone to error by anybody working on this code in the future, and b) avoids the code duplication you have in `detachDefault`.